src/pages/swap/index.tsx

```diff
@@ -109,14 +124,44 @@ export const SwapPage = () => {
       return;
     }
 
-    console.log('starting ....');
-
-    handleOnSubmit({
-      inputTokenType: from as InputTokenType,
-      outputTokenType: to as OutputTokenType,
-      amountInUnits: fromAmountString,
-      minAmountOutUnits: minimumOutputAmount.preciseString,
-    });
+    // test the route for starting token, then proceed
+    // will disable the confirm button
+    setIsInitiating(true);
+
+    const outputToken = OUTPUT_TOKEN_CONFIG[to];
+    const inputToken = INPUT_TOKEN_CONFIG[from];
+
+    // both route and stellar vault checks must be valid to proceed
+    const outputAmountBigMargin = Big(minimumOutputAmount.preciseString)
+      .round(2, 0)
+      .mul(1 + SPACEWALK_REDEEM_SAFETY_MARGIN); // add an X percent margin to be sure
+    const expectedRedeemAmountRaw = multiplyByPowerOfTen(outputAmountBigMargin, outputToken.decimals).toFixed();
+
+    const inputAmountBig = Big(fromAmountString);
```

Comment:
At this point `fromAmountString` should not be the empty string. However, if we change the logic and it is, then this will throw an error. Use `fromAmount` returned by `useSwapForm` instead.